framework/src/packets/icmp/v6/ndp/mod.rs

```diff
@@ -34,4 +37,47 @@ where
         let offset = self.payload_offset() + P::size();
         NdpOptionsIterator::new(mbuf, offset)
     }
+
+    fn add_option<T: NdpAddOptions>(&self, offset: usize) -> Result<T> {
```

Comment:
having add *and* push may be a bit confusing. Previously, @drunkirishcoder had something like parse that calls into do_parse, so maybe we want to use that convention too (and pick one of add or push).

Reply:
yep. agreed. i renamed them to be more consistent/clear